deepspeed/runtime/zero/stage_1_and_2.py

```diff
@@ -354,7 +366,11 @@ def __init__(self,
                 see_memory_usage(f"After Flattening and after emptying param group {i} cache", force=False)
 
             # set model bit16 weight to slices of flattened buffer
-            self._update_model_bit16_weights(i)
+            self._update_model_bit16_weights(i, use_cpu_data=True)
```

Comment:
Since the weight slices are already in a flattened buffer on device, I don't understand the `use_cpu_data` here.

Reply:
see my comment above